core/controllers/acl_decorators.py

```diff
@@ -2615,3 +2617,140 @@ def test_can_play_entity(self, entity_type, entity_id, **kwargs):
+
+        user_actions_info = user_services.UserActionsInfo(self.user_id)
+        if (
+                role_services.ACTION_ACCEPT_ANY_VOICEOVER_APPLICATION not in
+                user_actions_info.actions):
+            raise self.UnauthorizedUserException(
+                'You do not have credentials to review voiceover '
+                'applications.')
+
+        return handler(self, **kwargs)
+
+    test_can_review_voiceover_application.__wrapped__ = True
+
+    return test_can_review_voiceover_application
+
+
+def can_submit_voiceover_application(handler):
+    """Decorator to check whether a user can submit voiceover application.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now also checks
+            if the user has permission to submit voiceover application.
+    """
+
+    def test_can_submit_voiceover_application(self, **kwargs):
+        """Checks if the user can submit voiceover application.
+
+        Args:
+            **kwargs: *. Keyword arguments.
+
+        Returns:
+            *. The return value of the decorated function.
+
+        Raises:
+            UnauthorizedUserException: The user does not have
+                credentials to submit voiceover application.
+        """
+        if role_services.ACTION_SUBMIT_VOICEOVER_APPLICATION in (
+                self.user.actions):
+            # The voiceover_service returns user submitted voiceover
+            # applications in sorted order of the date of submission.
+            voiceover_applications = (
+                voiceover_services.get_user_submitted_voiceover_applications(
+                    self.user_id))
+            if voiceover_applications:
+                latest_voiceover_application = voiceover_applications[0]
+                if latest_voiceover_application.status == (
+                        suggestion_models.STATUS_IN_REVIEW):
+                    raise self.UnauthorizedUserException(
+                        'You can only submit one application at a time. '
+                        'It looks like your latest application is not yet '
+                        'reviewed.')
+                elif latest_voiceover_application.status == (
+                        suggestion_models.STATUS_ACCEPTED):
+                    if latest_voiceover_application.target_type == (
+                            suggestion_models.TARGET_TYPE_EXPLORATION):
+                        exploration = exp_fetchers.get_exploration_by_id(
+                            latest_voiceover_application.target_id)
+                        if exploration.has_complete_voiceover(
+                                latest_voiceover_application.language_code):
+                            return handler(self, **kwargs)
+                        else:
+                            raise self.UnauthorizedUserException(
+                                'You cannot submit a new application until the '
```

Comment:
This seems a bit odd. Once applied -- you don't need to submit another application again, right?

Remember we reframed this to be: submit the application once, if it's accepted, you can ask for assignments. We shouldn't use the word "submit application" to describe "asking for assignments".

Reply:
> submit the application once, if it's accepted, you can ask for assignments.

One application for one assignment, that's the way we are going. If for the given application someone else is assigned then they can select a different item to voiceover.)

I'm working on writing the requirement doc with the new changes, I'll share it with you once ready!

Reply:
That's not what I thought it would be. Per our discussion, I thought it would be: the first time you voiceover, you need to submit an application. If accepted, you can then subsequently request to be assigned to a lesson. If you finish a lesson (or withdraw from it), you can then request assignment to a different one.

Is this not the case?